### PR TITLE
Add requirements for container images to run with Nextflow

### PR DESCRIPTION
**Pull request by mribeirodantas**
Currently, there are no instructions on what to put in a container image to run with Nextflow in the containers section of the documentation. This Pull Request fixes this issue.

### CODE REVIEW

**Comment by bentsherman** [comment]
Well, these requirements are also listed under this section: https://nextflow.io/docs/latest/tracing.html#tasks

So it would be best to list them only once and link to it where necessary.

**Comment by mribeirodantas** [comment]
It makes sense. I will update the section on tracing accordingly.

**Comment by mribeirodantas** [comment]
Done, @bentsherman.

File: modules/nextflow/src/main/groovy/nextflow/container/CharliecloudBuilder.groovy [hunk 1]
```diff
@@ -46,6 +47,9 @@ class CharliecloudBuilder extends ContainerBuilder<CharliecloudBuilder> {
         if( params.containsKey('runOptions') )
             addRunOptions(params.runOptions.toString())
 
+        if( params.containsKey('readOnlyInputs') )
+            this.readOnlyInputs = params.readOnlyInputs?.toString() == 'true'
+
```
Comment (bentsherman) [inline modules/nextflow/src/main/groovy/nextflow/container/CharliecloudBuilder.groovy:52]: I think you got some unrelated commits in your branch, something might have messed up while merging from upstream

Comment (mribeirodantas) [reply modules/nextflow/src/main/groovy/nextflow/container/CharliecloudBuilder.groovy]: I just reverted all these unexpected changes.

File: docs/container.rst [hunk 2]
```diff
@@ -11,6 +11,11 @@ platform that supports a container runtime. Furthermore, the same pipeline can b
 executed with any of the supported container runtimes, depending on which runtimes are available
 in the target compute environment.
 
+.. note::
+  When creating your container image to use with Nextflow, be aware that Nextflow runs a background
+  process for each job in the target environment. Make sure ``bash`` and ``ps`` are installed in your
+  image, along with other tools required for collecting metrics (See the list :ref:`here <trace-required-packages>`).
```
Comment (pditommaso) [inline docs/container.rst:17]:
```suggestion
  When creating your container image to use with Nextflow make sure it includes Bash (3.x or later) and ``ps`` are installed in your
  image, along with other tools required for collecting metrics (See the list :ref:`here <trace-required-packages>`). Also, Bash should be available on the `/bin/bash` path, it should be the container entry-point. 
```

**Comment by mribeirodantas** [comment]
Thanks for the feedback, @pditommaso 😄

File: docs/container.rst [hunk 3]
```diff
@@ -11,6 +11,10 @@ platform that supports a container runtime. Furthermore, the same pipeline can b
 executed with any of the supported container runtimes, depending on which runtimes are available
 in the target compute environment.
 
+.. note::
+  When creating your container image to use with Nextflow make sure it includes Bash (3.x or later) and ``ps`` are installed in your
+  image, along with other tools required for collecting metrics (See the list :ref:`here <trace-required-packages>`). Also, Bash should be available on the `/bin/bash` path, it should be the container entry-point.
```
Comment (pditommaso) [inline docs/container.rst:16]:
```suggestion
  image, along with other tools required for collecting metrics (See the list :ref:`here <trace-required-packages>`). Also, Bash should be available on the path `/bin/bash` and it should be the container entry-point. 
```

**Review by pditommaso — APPROVED** [review]
